TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -15,626 +15,651 @@
+                    NetInfo.Lane laneInfo = lanes[laneIndex];
+                    if ((laneInfo.m_laneType & LaneConnectionManager.LANE_TYPES) != NetInfo.LaneType.None &&
+                        (laneInfo.m_vehicleType & LaneConnectionManager.VEHICLE_TYPES) != VehicleInfo.VehicleType.None) {
+
+                        Vector3? pos = null;
+                        bool isSource = false;
+                        bool isTarget = false;
+                        if (connManager.GetLaneEndPoint(segmentId, !isEndNode, laneIndex, laneId, laneInfo, out isSource, out isTarget, out pos)) {
+                            int sortedLaneIndex = sortedLanes.IndexOf(laneInfo);
+                            pos = (Vector3)pos + offset;
+                            float terrainY = Singleton<TerrainManager>.instance.SampleDetailHeightSmooth(((Vector3)pos));
+                            Vector3 finalPos = new Vector3(((Vector3)pos).x, terrainY, ((Vector3)pos).z);
+
+                            nodeMarkers.Add(new NodeLaneMarker() {
+                                segmentId = segmentId,
+                                laneId = laneId,
+                                nodeId = nodeId,
+                                startNode = !isEndNode,
+                                position = finalPos,
+                                secondaryPosition = (Vector3)pos,
+                                color = colors[nodeMarkers.Count % colors.Length],
+                                isSource = isSource,
+                                isTarget = isTarget,
+                                laneType = laneInfo.m_laneType,
+                                vehicleType = laneInfo.m_vehicleType,
+                                laneGroup = laneGroups[sortedLaneIndex],
+                                innerSimilarLaneIndex = ((byte)(laneInfo.m_direction & NetInfo.Direction.Forward) != 0) ? laneInfo.m_similarLaneIndex : laneInfo.m_similarLaneCount - laneInfo.m_similarLaneIndex - 1,
+                                segmentIndex = i
+                            });
+                        }
+                    }
+
+                    laneId = NetManager.instance.m_lanes.m_buffer[laneId].m_nextLane;
+                }
+            }
+
+            if (nodeMarkers.Count == 0)
+                return null;
+
+            foreach (NodeLaneMarker laneMarker1 in nodeMarkers) {
+                if (!laneMarker1.isSource)
+                    continue;
+
+                uint[] connections = LaneConnectionManager.Instance.GetLaneConnections(laneMarker1.laneId, laneMarker1.startNode);
+                if (connections == null || connections.Length == 0)
+                    continue;
+
+                foreach (NodeLaneMarker laneMarker2 in nodeMarkers) {
+                    if (!laneMarker2.isTarget)
+                        continue;
+                    if (laneMarker1.laneGroup != laneMarker2.laneGroup)
```

Comment:
currently, the lane group value cannot be compared between different segments

Reply:
You will ask: "what to do to make it comparable?"

It's a bit complicated...
```
NetInfo.Direction nextDir = isNextStartNodeOfNextSegment ? NetInfo.Direction.Backward : NetInfo.Direction.Forward;
				NetInfo.Direction nextDir2 = !nextSegIsInverted ? nextDir : NetInfo.InvertDirection(nextDir);
[...]
if ((nextLaneInfo.m_finalDirection & nextDir2) != NetInfo.Direction.None) { // next is incoming lane
[...]
}
```

Also check `RoutingManager.IsIncomingLane` and `IsOutgoingLane`.